src/server.c

```diff
@@ -3563,7 +3575,10 @@ void call(client *c, int flags) {
 
         /* Call alsoPropagate() only if at least one of AOF / replication
          * propagation is needed. */
-        if (propagate_flags != PROPAGATE_NONE) alsoPropagate(c->db->id, c->argv, c->argc, propagate_flags);
+        if (propagate_flags != PROPAGATE_NONE && shouldPropagate(propagate_flags)) {
+            alsoPropagate(c->db->id, c->argv, c->argc, propagate_flags);
+            c->cmd->flags |= CMD_PROPAGATE;
```

Comment:
The command `c->cmd` is metadata about the command and we shouldn't change it because it's used by all clients. Maybe we should set a client flag in `c->flags` instead?

```suggestion
            c->flags |= CLIENT_PROPAGATING;
```